app/internal/page/donor/register.go

```diff
@@ -329,3 +334,70 @@ func makeHandle(mux *http.ServeMux, store sesh.Store, defaultOptions handleOpt,
 		})
 	}
 }
+
+func makeLpaHandle(mux *http.ServeMux, store sesh.Store, defaultOptions handleOpt, errorHandler page.ErrorHandler, donorStore DonorStore, uidClient UidClient, logger Logger) func(string, handleOpt, Handler) {
+	return func(path string, opt handleOpt, h Handler) {
+		opt = opt | defaultOptions
+
+		mux.HandleFunc(path, func(w http.ResponseWriter, r *http.Request) {
+			ctx := r.Context()
+
+			appData := page.AppDataFromContext(ctx)
+			appData.Page = path
+			appData.CanGoBack = opt&CanGoBack != 0
+			appData.ActorType = actor.TypeDonor
+
+			donorSession, err := sesh.Login(store, r)
+			if err != nil {
+				http.Redirect(w, r, page.Paths.Start, http.StatusFound)
+				return
+			}
+
+			appData.SessionID = base64.StdEncoding.EncodeToString([]byte(donorSession.Sub))
+
+			sessionData, err := page.SessionDataFromContext(ctx)
+
+			if err == nil {
+				sessionData.SessionID = appData.SessionID
+				ctx = page.ContextWithSessionData(ctx, sessionData)
+
+				appData.LpaID = sessionData.LpaID
+			} else {
+				ctx = page.ContextWithSessionData(ctx, &page.SessionData{SessionID: appData.SessionID, LpaID: appData.LpaID})
+			}
+
+			lpa, err := donorStore.Get(ctx)
+			if err != nil {
+				errorHandler(w, r, err)
+				return
+			}
+
+			if lpa.Tasks.YourDetails == actor.TaskCompleted && lpa.UID == "" {
```

Comment:
it might also be good to have `r.Method == http.MethodPost` here, as it might be weird seeing an (unlikely) error from the donorStore write